xbout/load.py

```diff
@@ -317,6 +322,19 @@ def attrs_remove_section(obj, section):
         if info:
             warn("No geometry type found, no physical coordinates will be added")
 
+    if grid:
+        grididoptions = ds.metadata.get("grid_id", None)
+        grididfile = grid.get("grid_id", None)
+        if grididoptions and grididfile and grididfile != grididoptions:
+            msg = f"""The grid used for the simulation is not the one used.
+For the simulation {grididoptions} was used,
+but we did load {grididfile}."""
+            if grid_mismatch == "warn":
+                warn(msg)
+            elif gird_mismatch == "ignore":
+                pass
+            else:
+                raise RuntimeError(msg)
```

Comment:
Maybe `ValueError` would be better?
https://docs.python.org/3/library/exceptions.html#ValueError
```suggestion
                raise ValueError(msg)
```